src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -126,17 +127,18 @@ def _exec_batch(
             line_results = asyncio.run(self._exec_batch_internal(batch_inputs, output_dir, run_id))
         handle_line_failures([r.run_info for r in line_results], raise_on_line_failure)
         aggr_results = self._exec_aggregation_internal(batch_inputs, line_results, run_id)
+
+        # persist outputs to output dir
         outputs = [
             {LINE_NUMBER_KEY: r.run_info.index, **r.output}
             for r in line_results
             if r.run_info.status == Status.Completed
         ]
-        return BatchResult(
-            outputs=outputs,
-            metrics=aggr_results.metrics,
-            line_results=line_results,
-            aggr_results=aggr_results,
-        )
+        self._persist_outputs(outputs, output_dir)
```

Comment:
A question not related to this PR. How do we persist image in output.

Reply:
In line execution process pool
![image](https://github.com/microsoft/promptflow/assets/111329184/08d7027e-4527-4642-8191-5abdd8fcda00)